crates/bin/src/args.rs

```diff
@@ -88,9 +88,21 @@ pub struct Args {
     /// This skips searching crates.io for a manifest and uses the specified path directly, useful
     /// for debugging and when adding Binstall support. This may be either the path to the folder
     /// containing a Cargo.toml file, or the Cargo.toml file itself.
+    ///
+    /// This option cannot be used with `--git`.
```

Comment:
I understand why that is given the implementation, but wonder if we couldn't support that... clone the repo and 'override' the manifest essentially.

Reply:
I think we should do it via https://github.com/cargo-bins/cargo-binstall/pull/1162#discussion_r1236771442

For `--manifest-path`, it simply cannot specify the path to the cloned repository since it is a temporary path that only exists during cloning and loading `Cargo.toml`.

We would have to make `--manifest-path` a template for this to work, which IMHO is a bit too complicated.

Reply:
hmm I think if you specify both `--git` and `--manifest-path` you're going to have a relative or absolute path to a particular concrete file, so we don't need a template. If you only have `--git`, we can 'default' manifest-path to whatever we find in the repo.

Reply:
That is, the usecase of specifying both is not to point to a different subpath within the repo, but to have a completely different manifest for a git repo install, like for a crates install

Reply:
I think there'd still be a usecase but happy to keep this exclusive and punt until someone asks for it.

Reply:
> I understand why that is given the implementation, but wonder if we couldn't support that... clone the repo and 'override' the manifest essentially.

I've added support for handling workspace, so it should be ok even if users cannot override the manifest as long as the repo they cloned is a valid workspace.